datalake/src/main/java/com/sequenceiq/datalake/flow/datalake/scale/config/DatalakeHorizontalScaleFlowConfig.java

```diff
@@ -0,0 +1,94 @@
+import static com.sequenceiq.datalake.flow.datalake.scale.DatalakeHorizontalScaleState.DATALAKE_HORIZONTAL_SCALE_FINISHED_STATE;
+import static com.sequenceiq.datalake.flow.datalake.scale.DatalakeHorizontalScaleState.DATALAKE_HORIZONTAL_SCALE_SERVICES_RESTART_STATE;
+import static com.sequenceiq.datalake.flow.datalake.scale.DatalakeHorizontalScaleState.DATALAKE_HORIZONTAL_SCALE_START_STATE;
+import static com.sequenceiq.datalake.flow.datalake.scale.DatalakeHorizontalScaleState.DATALAKE_HORIZONTAL_SCALE_VALIDATION_STATE;
+import static com.sequenceiq.datalake.flow.datalake.scale.DatalakeHorizontalScaleState.INIT_STATE;
+
+import java.util.List;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.datalake.flow.datalake.scale.DatalakeHorizontalScaleEvent;
+import com.sequenceiq.datalake.flow.datalake.scale.DatalakeHorizontalScaleState;
+import com.sequenceiq.flow.core.config.AbstractFlowConfiguration;
+import com.sequenceiq.flow.core.config.AbstractFlowConfiguration.Transition.Builder;
+import com.sequenceiq.flow.core.config.RetryableFlowConfiguration;
+
+@Component
+public class DatalakeHorizontalScaleFlowConfig extends AbstractFlowConfiguration<DatalakeHorizontalScaleState, DatalakeHorizontalScaleEvent>
+        implements RetryableFlowConfiguration<DatalakeHorizontalScaleEvent> {
+
+    private static final List<Transition<DatalakeHorizontalScaleState, DatalakeHorizontalScaleEvent>> TRANSITIONS =
+            new Builder<DatalakeHorizontalScaleState, DatalakeHorizontalScaleEvent>()
+                    .defaultFailureEvent(DATALAKE_HORIZONTAL_SCALE_FAILED_EVENT)
+
+                    .from(INIT_STATE)
+                    .to(DATALAKE_HORIZONTAL_SCALE_VALIDATION_STATE)
+                    .event(DATALAKE_HORIZONTAL_SCALE_EVENT)
+                    .defaultFailureEvent()
+
+                    .from(DATALAKE_HORIZONTAL_SCALE_VALIDATION_STATE)
+                    .to(DATALAKE_HORIZONTAL_SCALE_START_STATE)
+                    .event(DATALAKE_HORIZONTAL_SCALE_START_EVENT)
+                    .defaultFailureEvent()
+
+                    .from(DATALAKE_HORIZONTAL_SCALE_START_STATE)
+                    .to(DATALAKE_HORIZONTAL_SCALE_SERVICES_RESTART_STATE)
+                    .event(DATALAKE_HORIZONTAL_SCALE_CM_ROLLING_RESTART_EVENT)
+                    .failureEvent(DATALAKE_HORIZONTAL_SCALE_CM_RESTART_FAILED_EVENT)
+
+                    .from(DATALAKE_HORIZONTAL_SCALE_SERVICES_RESTART_STATE)
+                    .to(DATALAKE_HORIZONTAL_SCALE_FINISHED_STATE)
+                    .event(DATALAKE_HORIZONTAL_SCALE_FINISHED_EVENT)
+                    .noFailureEvent()
+
+                    .build();
+
+    private static final FlowEdgeConfig<DatalakeHorizontalScaleState, DatalakeHorizontalScaleEvent> EDGE_CONFIG =
+            new FlowEdgeConfig<>(INIT_STATE, DATALAKE_HORIZONTAL_SCALE_FINISHED_STATE, DATALAKE_HORIZONTAL_SCALE_FAILED_STATE,
```

Comment:
Usually we don't use a custom state as the final state, since this can result in weird conditions if an error occurs. I recommend following what we do in other flows for this.